### ISSUE TITLE
Support quickly flipping transaction/posting flag between "*" and "!"

### ISSUE THREAD

**Issue by pmarciniak** [beancount/fava#1151]
Flipping the flag by clicking on it directly in "Account Journal" view (without having to enter the transaction edit mode) would allow quickly reconciling the ledger against a bank statement, like it's possible e.g. in GnuCash.

**Comment by yagebu** [comment]
A PR would be welcome :)

**Comment by GuillaumeHM** [comment]
Hello,

I'm coming from Grisbi accounting program and the feature you describe is the exact one I **REALLY** miss.

As a workaround, I developed a vim function to toggle flag with a mouse click. But a ledger embeds all transactions whereas fava filters on the only account (with a clean tx list display) that is much more ergonomic to do reconciliation with the bank tx list. For now, I need to reconcile in the ledger file, then check in the fava's account display that not a transaction remains unreconciled before the balance entry.

So currently, I need to switch between vim and fava windows and between fava's pages to check the balance error amount in the error page (see #1766). This is very painful regarding any accounting program that handles this basic case much better (all in one view: balance error, clickable transaction's flag).

My web development knowledge reaches almost zero, so I can't do it myself. Can any one develop a small plugin to achieve that ? Please please please !

I'm open to discuss about the ergonomic aspect of this feature (reconcile mode, keyboard + click combo...) and to help tweaking, testing...

Thank you very much in advance !

**Comment by GuillaumeHM** [comment]
Hello,

Back to this subject which I still miss a lot, and it seems I'm not the only one since there are at least three issues on this topic. I saw in #456 that @yagebu told that Entry editor should give a simple way to do that. But this solution would still need to open and close the editor. That means double or triple clicks to flip each flag.

I don't clearly understand where is the difficulty to implement this feature since there yet is code to change any transaction in the file. Isn't it possible to re-use some editor functions to flip the flag without displaying the editor itself ? The keybord shortcut + mouse click approach seemed good...

Thank you !

**Comment by yagebu** [comment]
> I don't clearly understand where is the difficulty to implement this feature since there yet is code to change any transaction in the file.

I agree, it shouldn't be too hard to impelement - as written above, a PR implementing it would be welcome :)

**Comment by GuillaumeHM** [comment]
I known... But unfortunately, my web coding skills almost stops at HTML/CSS... So I rely on others for this one :(

**Comment by blalor** [comment]
I used an agent write an extension to add this functionality: https://github.com/blalor/fava-flagger 

@yagebu I can rework it into a PR if you'd like, but feel free to take the implementation from the extension.

**Comment by GuillaumeHM** [comment]
@blalor, thank you for that plugin !

I tried it without success. But it's probably due to that old fava version I'm stuck in (1.29 which is latest compatible with beancount v2).

My fava is installed via `pipx` so I used `pipx inject fava git+https://github.com/blalor/fava-flagger.git`. I didn't run the `uv` command indicated in your README because I didn't know how to do it and I thunk it could luckily work without... Should I have ?

Note: Strangely, I had a fava error on a parameter line which was working when the plugin wasn't installed : `2020-01-01 custom "fava-option" "unrealized" "Plus-values-latentes"`. Maybe a conflict somewhere ? Can you test with an up to date fava installation ?

Again, this experience report from my old fava version might not be of interest. I'm thinking to upgrade someday...
